Extension/src/Utility/Async/factory.ts

```diff
@@ -0,0 +1,55 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+/* eslint-disable @typescript-eslint/naming-convention */
+
+import { is } from '../System/guards';
+import { Reject, Resolve } from '../System/types';
+
+type WithInit<TOutput> = TOutput & { init?(...args: any[]): any | Promise<any> };
+
+export function Factory<TFactory extends (...args: Parameters<TFactory>) => ReturnType<TFactory>>(factory: TFactory): new (...args: Parameters<TFactory>) => Promise<Awaited<ReturnType<TFactory>>> {
+    /** calls/awaits any .init in the result from the factory */

```

Comment:
As mentioned earlier, why is this comment style used here?

Reply:
This is used as a documentation comment